File: backend/metier_generator/templates/threed/api/controllers/asset_controller.js
// Contrôleur métier threed (Node.js)
const assets = [];
module.exports = {
  listAssets: (req, res) => res.json(assets),
  createAsset: (req, res) => {
    const asset = req.body;
    assets.push(asset);
    res.json(asset);
  },
  getAsset: (req, res) => {
    const asset = assets.find(a => a.id == req.params.id);
    res.json(asset || {});
  },
  deleteAsset: (req, res) => {
    const idx = assets.findIndex(a => a.id == req.params.id);
    if (idx !== -1) assets.splice(idx, 1);
    res.json({ deleted: idx !== -1 });
  }
};

File: backend/metier_generator/templates/threed/api/controllers/asset_controller.py
"""
Contrôleur métier threed (Python)
"""


class AssetController:
    assets = []

    @staticmethod
    def list_assets():
        return AssetController.assets

    @staticmethod
    def create_asset(data):
        AssetController.assets.append(data)
        return data

    @staticmethod
    def get_asset(id):
        for a in AssetController.assets:
            if a.get("id") == id:
                return a
        return {}

    @staticmethod
    def delete_asset(id):
        before = len(AssetController.assets)
        AssetController.assets = [
            a for a in AssetController.assets if a.get("id") != id
        ]
        return {"deleted": len(AssetController.assets) < before}
